src/compiler/checker.ts

```diff
@@ -22962,9 +22962,16 @@ namespace ts {
                                 case SyntaxKind.AsteriskAsteriskToken: return left ** right;
                             }
                         }
+                        if (typeof left === "string" && typeof right === "string") {
+                            switch ((<BinaryExpression>expr).operatorToken.kind) {

```

Comment:
i would make this a simple if statement.. and consider merging it with `typeof left === "string" && typeof right === "string`.